src/inspector_agent.cc

```diff
@@ -554,20 +564,25 @@ void Agent::Stop() {
   if (io_ != nullptr) {
     io_->Stop();
     io_.reset();
-    enabled_ = false;
   }
 }
 
-void Agent::Connect(InspectorSessionDelegate* delegate) {
-  enabled_ = true;
-  client_->connectFrontend(delegate);
+std::unique_ptr<InspectorSession> Agent::Connect(
+    std::unique_ptr<InspectorSessionDelegate> delegate) {
+  int session_id = client_->connectFrontend(std::move(delegate));
+  return std::make_unique<InspectorSession>(session_id, client_.get());
 }
 
 void Agent::WaitForDisconnect() {
   CHECK_NE(client_, nullptr);
   client_->contextDestroyed(parent_env_->context());
   if (io_ != nullptr) {
     io_->WaitForDisconnect();
+    // There is a bug in inspector that causes every session
+    // call V8InspectorClient::quitMessageLoopOnPause. We are using this flag
+    // to ignore those calls so message loop is spinning as long as there's a
+    // reason to expect inspector messages
```

Comment:
Is there a crbug for this?

Reply:
https://crbug.com/834056

Reply:
Can you put that link in the comment?

Reply:
I added it in a separate commit, will squash before merging.